### PR TITLE
Refactor TextLineDatasetOp with tests

### PR DESCRIPTION
**Pull request by feihugis**
This PR refactors `TextLineDatasetOp` using name_utils and add the tests.

cc: @jsimsa

### CODE REVIEW

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 1]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
 #include "tensorflow/core/util/ptr_util.h"
 
+#include "tensorflow/core/lib/io/zlib_compression_options.h"
+#include "tensorflow/core/lib/io/zlib_outputbuffer.h"
+#include "tensorflow/core/platform/env.h"
+
 namespace tensorflow {
 namespace data {
 
+enum CompressionType : int { ZLIB = 0, GZIP = 1, RAW = 2, UNCOMPRESSED = 3 };
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: if you use `enum class` you will not need to worry about unsupported enum values.

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 2]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
+
+// Gets the compression name.
+string CompressionName(CompressionType compression_type);
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: Rename the method to `CompressionTypeToString` (or simply `ToString`) and update the documentation to "Returns a string representation for the given compression type."`

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 3]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
+
+// Gets the specified zlib compression options according to the compression
+// type. Note that `CompressionType : UNCOMPRESSED` is not supported because
+// `ZlibCompressionOptions` does not have the option for uncompressed data.
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: does not have *an* option

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 4]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
 namespace tensorflow {
 namespace data {
 
+enum CompressionType : int { ZLIB = 0, GZIP = 1, RAW = 2, UNCOMPRESSED = 3 };
+
+// Gets the compression name.
+string CompressionName(CompressionType compression_type);
+
+// Gets the specified zlib compression options according to the compression
+// type. Note that `CompressionType : UNCOMPRESSED` is not supported because
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: `CompressionType::UNCOMPRESSED`

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 5]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
+// `ZlibCompressionOptions` does not have the option for uncompressed data.
+io::ZlibCompressionOptions GetZlibCompressionOptions(
+    CompressionType compression_type);
+
+// Saves the input text into the file with the specified compression.
+Status CreateFile(const string& filename, const std::vector<string>& text,
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: why is the input data a vector of string? that seems to be an awkward API

Document what is `input_buffer_size` and `output_buffer_size` for.

Rename the method to `WriteDataToFile` and change data to be `char *` (unless there is a good reason you would want it to be vector of string).

File: tensorflow/core/kernels/data/dataset_test_base.h [hunk 6]
```diff
@@ -38,9 +38,35 @@ limitations under the License.
...
+                  int input_buffer_size, int output_buffer_size,
+                  const CompressionType& compression_type);
+
+// Saves the input texts into multiple files with the specified compression.
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/dataset_test_base.h]: Similar comments as above. In addition, document what portion of `multi_texts` is stored in what files?

What is the purpose of having this method as opposed to calling `WriteDataToFile` in a loop?

**Review by jsimsa — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 1 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Good suggestion! Changed it.

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 2 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Done

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 3 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Done

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 4 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Done

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 5 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Thanks for your suggestions! Yes, this API is not good. Have changed it as you suggested. Also add the docs for `input_buffer_size` and `output_buffer_size`.

File: tensorflow/core/kernels/data/dataset_test_base.h (hunk 6 above)
Comment (feihugis) [reply tensorflow/core/kernels/data/dataset_test_base.h]: Changed the test case to specify which file stores which text, and this function is removed. Now we call `WriteDataToFile` in a loop to create multiple text files. Thanks for the suggestion!

**Comment by feihugis** [comment]
@jsimsa Thanks for your review! The comments are addressed in this commit (https://github.com/tensorflow/tensorflow/pull/30540/commits/b872edc887833e3ae4d91dcd01df014861cb2cc6). Could you please have a look at the change when you get a chance?

**Review by jsimsa — CHANGES_REQUESTED** [review]
thanks, left a couple of comments on https://github.com/tensorflow/tensorflow/commit/b872edc887833e3ae4d91dcd01df014861cb2cc6

**Comment by feihugis** [comment]
@jsimsa The comments are addressed here(https://github.com/tensorflow/tensorflow/pull/30540/commits/7df87194e2335b33b8085ae06fb5f4f3fbef3ea9). Please take another look!

**Comment by feihugis** [comment]
@jsimsa The default values for `CompressionParams` are added now (https://github.com/tensorflow/tensorflow/pull/30540/commits/f6e973eb9ffaac771c57fac24ab03cdb658343f0).

**Review by jsimsa — APPROVED** [review]

**Comment by feihugis** [comment]
@jsimsa The internal check failed. Could you please help check the logs and paste them here?

**Comment by rthadur** [comment]
> @jsimsa The internal check failed. Could you please help check the logs and paste them here?

@feihugis changes are submitted internally , waiting for tool to auto merge , thank you

**Comment by feihugis** [comment]
> @feihugis changes are submitted internally , waiting for tool to auto merge , thank you

Thanks @rthadur! When I checked the tests, the status shows `failed`, which may be caused by that the test is still running. Now it is updated to `passed`.

**Pull request merged**